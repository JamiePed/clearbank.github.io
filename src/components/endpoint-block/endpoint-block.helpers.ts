import { sample } from 'openapi-sampler'

import { SectionType, CodeblockType } from 'src/global.types'

import { SectionTypeEnum, BlockStatusColourEnum } from './endpoint-block.types'

const prettyPrintJson = (payload:JSON):string => {
  return JSON.stringify(payload, null, 2)
}

const getCardColour = (payload: any):string => {
  return BlockStatusColourEnum[payload.description] || BlockStatusColourEnum.Default
}

const getCardTitle = (section: SectionType, payload: any):string => {
  if (section === SectionTypeEnum.RESPONSE) {
    return payload.description
  }

  return section
}

const createCodeBlock = (section: SectionType, payload: any):CodeblockType => {
  const content = sample(payload.content['application/json'].schema)

  return {
    title: getCardTitle(section, payload),
    section,
    color: getCardColour(payload),
    language: 'json',
    codeSnippet: prettyPrintJson(content)
  }
}

export const generateDefaultCodeblocks = (section:SectionType, content:any):CodeblockType[] => {
  if (!content) {
    return null
  }

  const codeblocks:CodeblockType[] = []

  if (section === SectionTypeEnum.REQUEST) {
    codeblocks.push(createCodeBlock(section, content))
  } else if (section === SectionTypeEnum.RESPONSE) {
    const responseCodes = Object.values(content)

    for (const response of responseCodes) {
      if (!response.hasOwnProperty('content')) continue
      codeblocks.push(createCodeBlock(section, response))
    }
  }

  return codeblocks
}

export const getWebhook = (webhooks: any, fileName: string) => {
  const { node } = webhooks.find(
    (webhook: any) => webhook.node.fields.slug === fileName
  )

  return node
}
