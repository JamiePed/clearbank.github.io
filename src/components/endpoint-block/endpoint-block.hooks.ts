import { useStaticQuery, graphql } from 'gatsby'

const PathRequestsQuery: React.FunctionComponent<any> = (): any => {
  const version = '_v1_Accounts'
  const query = graphql`
    query ResponsesQuery {
      allDataJson {
        edges {
          node {
            paths {
              _v1_Accounts {
                get {
                  produces
                  tags
                  responses {
                    _503 {
                      description
                      headers {
                        X_Correlation_Id {
                          description
                          type
                        }
                      }
                    }
                    _500 {
                      description
                    }
                    _403 {
                      description
                    }
                    _200 {
                      description
                      schema {
                        _ref
                      }
                      headers {
                        X_Correlation_Id {
                          description
                          type
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  `

  // return new Promise((resolve, reject) =>
  //   resolve(
  //     graphql(`{${theQuery}}`).then(result => {
  //       if (result.errors) {
  //         console.error(result.errors);

  //         reject(result.errors);
  //       }

  //       return result;
  //     })
  //   )
  // );

  // const query = useStaticQuery(graphql(theQuery));

  return query
}

export default PathRequestsQuery
