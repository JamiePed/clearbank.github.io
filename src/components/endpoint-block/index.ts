export { default } from './endpoint-block'
