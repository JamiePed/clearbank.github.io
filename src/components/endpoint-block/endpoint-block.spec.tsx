import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './endpoint-block'

import * as Types from './endpoint-block.types'

let component: ReturnType<typeof render> = null
let fragment: DocumentFragment = null
let root: HTMLElement = null

const defaultProps: Types.EndpointProps = {
  title: 'title2',
  path: 'path',
  type: 'type',
  boxProps: {},
  endpoints: []
}

beforeEach(() => {
  component = render(<Component {...defaultProps} children='children' />)

  fragment = component.asFragment()
  root = component.container.firstChild
})

afterEach(() => {
  fragment = null
  root = null
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(root).toMatchSnapshot()
  })
})
