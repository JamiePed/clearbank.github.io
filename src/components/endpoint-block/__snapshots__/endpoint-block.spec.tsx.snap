// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `null`;
